File: app/src/main/java/com/bryan/platform/activity/MainActivity.java
package com.bryan.platform.activity;

import android.content.Intent;
import android.os.Bundle;
import android.util.Log;
import android.view.View;
import android.widget.Toast;

import androidx.annotation.NonNull;
import androidx.appcompat.app.AppCompatActivity;
import androidx.recyclerview.widget.DividerItemDecoration;
import androidx.recyclerview.widget.LinearLayoutManager;

import com.bryan.platform.R;
import com.bryan.platform.activity.user.ProfileActivity;
import com.bryan.platform.adapter.MomentAdapter;
import com.bryan.platform.databinding.ActivityMainBinding;
import com.bryan.platform.model.entity.Moment;
import com.bryan.platform.model.response.Result;
import com.bryan.platform.model.response.SpringPage;
import com.bryan.platform.network.MomentService;
import com.bryan.platform.network.RetrofitClient;

import java.util.Collections;

import retrofit2.Call;
import retrofit2.Callback;
import retrofit2.Response;

public class MainActivity extends AppCompatActivity {

    private ActivityMainBinding binding;
    private MomentAdapter momentAdapter;
    private final MomentService momentService =
            RetrofitClient.getInstance().createService(MomentService.class);

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        binding = ActivityMainBinding.inflate(getLayoutInflater());
        setContentView(binding.getRoot());

        setSupportActionBar(binding.toolbar);

        setupRecyclerView();
        fetchMoments();
        setupBottomNavigation();

        binding.fabNewPost.setOnClickListener(v ->
                Toast.makeText(this, "Create new post clicked!", Toast.LENGTH_SHORT).show()
        );
    }

    private void setupRecyclerView() {
        momentAdapter = new MomentAdapter(Collections.emptyList());
        binding.rvMoments.setAdapter(momentAdapter);
        binding.rvMoments.setLayoutManager(new LinearLayoutManager(this));
        binding.rvMoments.addItemDecoration(
                new DividerItemDecoration(this, DividerItemDecoration.VERTICAL)
        );
    }

    private void fetchMoments() {
        binding.progressBar.setVisibility(View.VISIBLE);
        momentService.getAllMoments(1, 10, "DESC").enqueue(new Callback<Result<SpringPage<Moment>>>() {
            @Override
            public void onResponse(@NonNull Call<Result<SpringPage<Moment>>> call,
                                   @NonNull Response<Result<SpringPage<Moment>>> response) {
                binding.progressBar.setVisibility(View.GONE);
                if (response.isSuccessful()) {
                    Result<SpringPage<Moment>> result = response.body();
                    if (result != null && result.isSuccess()) {
                        SpringPage<Moment> page = result.getData();
                        momentAdapter.updateData(
                                page != null ? page.getContent() : Collections.emptyList()
                        );
                        Log.d("MainActivity",
                                "Successfully fetched " +
                                        (page != null ? page.getContent().size() : 0) + " moments.");
                    } else {
                        String msg = result != null ? result.getMessage() : "Failed to fetch moments";
                        Toast.makeText(MainActivity.this, msg, Toast.LENGTH_LONG).show();
                        Log.e("MainActivity", "API Error: " + msg);
                    }
                } else {
                    String msg = "HTTP Error: " + response.code();
                    Toast.makeText(MainActivity.this, msg, Toast.LENGTH_LONG).show();
                    Log.e("MainActivity", msg);
                }
            }

            @Override
            public void onFailure(@NonNull Call<Result<SpringPage<Moment>>> call,
                                  @NonNull Throwable t) {
                binding.progressBar.setVisibility(View.GONE);
                String msg = "Network Failure: " + t.getMessage();
                Toast.makeText(MainActivity.this, msg, Toast.LENGTH_LONG).show();
                Log.e("MainActivity", msg, t);
            }
        });
    }

    private void setupBottomNavigation() {
        binding.bottomNavigation.setOnItemSelectedListener(item -> {
            int id = item.getItemId();
            if (id == R.id.navigation_home || id == R.id.navigation_dashboard) {
                // 当前页，无需跳转
                return true;
            } else if (id == R.id.navigation_profile) {
                startActivity(new Intent(MainActivity.this, ProfileActivity.class));
                return true;
            }
            return false;
        });

        // 默认选中
        binding.bottomNavigation.setSelectedItemId(R.id.navigation_dashboard);
    }
}

File: app/src/main/java/com/bryan/platform/activity/MainActivity.kt
package com.bryan.platform.activity

import android.content.Intent
import android.os.Bundle
import android.util.Log
import android.view.View
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import androidx.recyclerview.widget.DividerItemDecoration
import androidx.recyclerview.widget.LinearLayoutManager
import com.bryan.platform.R
import com.bryan.platform.adapter.MomentAdapter
import com.bryan.platform.databinding.ActivityMainBinding
import com.bryan.platform.model.response.SpringPage
import com.bryan.platform.model.entity.Moment
import com.bryan.platform.model.response.Result
import com.bryan.platform.network.MomentService
import com.bryan.platform.network.RetrofitClient
import retrofit2.Call
import retrofit2.Callback
import retrofit2.Response

class MainActivity : AppCompatActivity() {

    private lateinit var binding: ActivityMainBinding
    private lateinit var momentAdapter: MomentAdapter
    private val momentService: MomentService = RetrofitClient.createService(MomentService::class.java)

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivityMainBinding.inflate(layoutInflater)
        setContentView(binding.root)

        setSupportActionBar(binding.toolbar)

        setupRecyclerView()
        fetchMoments()
        setupBottomNavigation() // Setup bottom navigation

        binding.fabNewPost.setOnClickListener {
            // TODO: Implement logic to open a new screen for creating a post
            Toast.makeText(this, "Create new post clicked!", Toast.LENGTH_SHORT).show()
        }
    }

    private fun setupRecyclerView() {
        momentAdapter = MomentAdapter(emptyList())
        binding.rvMoments.apply {
            adapter = momentAdapter
            layoutManager = LinearLayoutManager(this@MainActivity)
            // Add a divider line between items
            addItemDecoration(DividerItemDecoration(this@MainActivity, DividerItemDecoration.VERTICAL))
        }
    }

    private fun fetchMoments() {
        binding.progressBar.visibility = View.VISIBLE
        momentService.getAllMoments().enqueue(object : Callback<Result<SpringPage<Moment>>> {
            override fun onResponse(call: Call<Result<SpringPage<Moment>>>, response: Response<Result<SpringPage<Moment>>>) {
                binding.progressBar.visibility = View.GONE
                if (response.isSuccessful) {
                    val result = response.body()
                    if (result != null && result.isSuccess()) {
                        val moments = result.data?.content ?: emptyList()
                        momentAdapter.updateData(moments)
                        Log.d("MainActivity", "Successfully fetched ${moments.size} moments.")
                    } else {
                        val errorMsg = result?.message ?: "Failed to fetch moments"
                        Toast.makeText(this@MainActivity, errorMsg, Toast.LENGTH_LONG).show()
                        Log.e("MainActivity", "API Error: $errorMsg")
                    }
                } else {
                    val errorMsg = "HTTP Error: ${response.code()}"
                    Toast.makeText(this@MainActivity, errorMsg, Toast.LENGTH_LONG).show()
                    Log.e("MainActivity", errorMsg)
                }
            }

            override fun onFailure(call: Call<Result<SpringPage<Moment>>>, t: Throwable) {
                binding.progressBar.visibility = View.GONE
                val errorMsg = "Network Failure: ${t.message}"
                Toast.makeText(this@MainActivity, errorMsg, Toast.LENGTH_LONG).show()
                Log.e("MainActivity", errorMsg, t)
            }
        })
    }

    private fun setupBottomNavigation() {
        binding.bottomNavigation.setOnItemSelectedListener { item ->
            when (item.itemId) {
                // R.id.navigation_home is the current activity, so no action needed or you can re-fetch data
                R.id.navigation_home -> {
                    // Optionally re-fetch moments or just stay on this screen
                    true
                }
                R.id.navigation_dashboard -> {
                    // This is also the current activity (Moments/Dashboard)
                    // No action needed, or you can ensure it's the top of the stack
                    true
                }
                R.id.navigation_profile -> {
                    // Navigate to ProfileActivity
                    val intent = Intent(this, ProfileActivity::class.java)
                    startActivity(intent)
                    true
                }
                else -> false
            }
        }
        // Set the default selected item to match the current activity
        binding.bottomNavigation.selectedItemId = R.id.navigation_dashboard // Assuming MainActivity is the Dashboard/Moments screen
    }
}
